packages/hardhat-core/src/internal/core/jsonrpc/types/base-types.ts

```diff
@@ -29,6 +29,41 @@ export const rpcHash = new t.Type<Buffer>(
   t.identity
 );
 
+export const rpcStorageSlot = new t.Type<BN>(
+  "Storage slot",
+  BN.isBN,
+  validateStorageSlot,
+  t.identity
+);
+
+function validateStorageSlot(u: unknown, c: t.Context): t.Validation<BN> {
+  if (typeof u !== "string") {
+    return t.failure(
+      u,
+      c,
+      `Storage slot argument must be a string, got '${u}'`
+    );
+  }
+
+  if (u.match(/^0x(?:[0-9a-fA-F]*)*$/) === null) {
+    return t.failure(
+      u,
+      c,
+      `Storage slot argument must be a valid hexadecimal, got '${u}'`
+    );
+  }
+
+  if (u.length !== 66) {
+    return t.failure(
+      u,
+      c,
+      `Storage slot argument must have a length of 66 (32 bytes), but '${u}' has a length of ${u.length}`
```

Comment:
I added more useful and specific error messages here to help people that are using this method with shorter values, which we were (wrongly) accepting before.

Reply:
I think you could also expand this a little more, just to be a little more educational:
```suggestion
      `Storage slot argument must have a length of 66 (32 bytes (2 characters per byte) plus the "0x" prefix), but '${u}' has a length of ${u.length}`
```

Reply:
```suggestion
      `Storage slot argument must have a length of 66 ("0x" + 32 bytes), but '${u}' has a length of ${u.length}`
```